docs/Testing.md

```diff
@@ -0,0 +1,132 @@
+# Testing
+
+A test case takes the form of a binary \<basename\>.wasm file, next to that
+that there will always be a \<basename\>.\<ext\> source file from which the
+test was originally compiled from which can be used as a reference in the event
+of an error.
+
+Additionally, any of the following optional auxilary files and directories may
+be present:
+- `<basename>.arg`
+- `<basename>.env`
+- `<basename>.dir`
+- `<basename>.stdin`
+- `<basename>.stdout`
+- `<basename>.stderr`
+- `<basename>.status`
+
+## Running tests
+
+To run a test do the following steps to prepare for execution:
+
+- Prepare inputs
+  - Given an `<input>.wasm` file; take the `<basename>` of said file.
+  - If `<basename>.<arg>` exists; take the program arguments from said file.
+  - If `<basename>.<env>` exists; take the program environment from said file.
+  - If `<basename>.<dir>` exists; preopen the directory from said file.
+  - If `<basename>.<stdin>` exists; pipe said file into the program as stdin.
+- Run program
+- Collect results
+  - If `<basename>.<stdout>` exists; assert that the programs stdout matches
+    said file.
+  - If `<basename>.<stderr>` exists; assert that the programs stdout matches
+    said file.
+  - If `<basename>.<status>` exists; assert that the programs stdout matches
+    said file; otherwise assert that the program exited with status 0.
+- Pass
+
+For example:
+
+```bash
+# Usage: $1 <runtime> <path_to_binary.wasm>
+# $1 wasmtime proc_exit-success.wasm
+# $1 wasmer proc_exit-failure.wasm
```

Comment:
I know these are only examples, but in the interests of not wanting to exclude engines, we shouldn't mention either of these examples here. Would "Usage: $1 <runtime_cmd> <path_to_binary>.wasm" be clear enough, without the examples?